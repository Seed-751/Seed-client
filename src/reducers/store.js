import { configureStore } from "@reduxjs/toolkit";
import createSagaMiddleware from "@redux-saga/core";
import logger from "redux-logger";

import { rootSaga } from "../sagas/rootSaga";

import userReducer from "./userSlice";
import musicReducer from "./musicSlice";

const sagaMiddleware = createSagaMiddleware();

const middleware = process.env.NODE_ENV === "development"
  ? [logger, sagaMiddleware]
  : (getDefaultMiddleware) => getDefaultMiddleware().concat(sagaMiddleware);

const store = configureStore({
  reducer: {
    user: userReducer,
    music: musicReducer,
  },
  middleware,
});

sagaMiddleware.run(rootSaga);

export default store;
