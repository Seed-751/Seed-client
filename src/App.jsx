import React, { useState, useEffect } from "react";
import { Route, Switch, Redirect } from "react-router-dom";
import { useSelector, useDispatch } from "react-redux";
import styled, { ThemeProvider } from "styled-components";

import { authCheckRequest, selectUser } from "./reducers/userSlice";
import { selectError } from "./reducers/errorSlice";

import Header from "./components/Header";
import Navigation from "./components/Navigation";
import PrivateRoute from "./components/PrivateRoute";
import Modal from "./components/Modal/Modal";
import Payment from "./pages/Payment";
import Error from "./components/Modal/Error";
import Login from "./pages/Login";
import Signup from "./pages/Signup";
import Dashboard from "./pages/Dashboard";
import MusicDetail from "./pages/MusicDetail";
import MyPage from "./pages/MyList";
import Upload from "./pages/Upload";

import GlobalStyles from "./styles";
import theme from "./styles/theme";

const Section = styled.section`
  display: flex;
  flex-direction: column;
  width: calc(100% - 230px);
  min-height: 100vh;
`;

const Container = styled.div`
  display: flex;
`;

export default function App() {
  const [isLoaded, setIsLoaded] = useState(false);
  const { userInfo, isLoggedIn } = useSelector(selectUser);
  const { isError, message } = useSelector(selectError);
  const dispatch = useDispatch();

  useEffect(() => {
    if (isLoggedIn) {
      return handleLoaded(true);
    }

    const token = localStorage.getItem("token");

    console.log(token);
    if (!token) {
      return handleLoaded(true);
    }

    dispatch(authCheckRequest(token));

    // eslint-disable-next-line
  }, [isLoggedIn]);

  function handleLoaded(value) {
    setIsLoaded(value);
  }

  return (
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      {isLoaded &&
        <>
          <Container>
            {isError &&
              <Modal isError={isError}>
                <Error message={message} />
              </Modal>
            }
            <Navigation />
            <Section>
              <Header userInfo={userInfo} />
              <Switch>
                <Route path="/login">
                  <Login />
                </Route>
                <Route path="/signup">
                  <Signup />
                </Route>
                <Route path="/dashboard">
                  <Dashboard />
                </Route>
                <Route path="/musics/:music_id">
                  <MusicDetail />
                </Route>
                <PrivateRoute
                  path="/upload"
                  isAuthenticated={isLoggedIn}
                  component={Upload}
                />
                <PrivateRoute
                  path="/mypage"
                  isAuthenticated={isLoggedIn}
                  component={MyPage}
                />
                <PrivateRoute
                  path="/payment/:albumId/:userId"
                  isAuthenticated={isLoggedIn}
                  component={Payment}
                />
                <Route path="/" exact>
                  <Redirect to="/dashboard" />
                </Route>
                <Route path="*">
                  <Redirect to="/dashboard" />
                </Route>
              </Switch>
            </Section>
          </Container>
        </>
      }
    </ThemeProvider>
  );
}
