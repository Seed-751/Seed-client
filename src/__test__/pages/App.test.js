import React from "react";
import { Provider } from "react-redux";
import { render, screen } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { BrowserRouter as Router } from "react-router-dom";

import store from "../../reducers/store";
import App from "../../App";

describe("<App />", () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  it("matches snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it("should render Login, Signup Button if user not logged in", () => {
    window.IntersectionObserver = jest.fn(() => ({
      observe: jest.fn(),
      disconnect: jest.fn(),
    }));

    const history = createMemoryHistory();

    render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );

    history.push("/");

    expect(screen.getByText("Login")).toBeInTheDocument();
    expect(screen.getByText("Signup")).toBeInTheDocument();
  });
});
