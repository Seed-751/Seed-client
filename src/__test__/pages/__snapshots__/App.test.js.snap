// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div>
  <div
    class="sc-crrszt fRjkOT"
  >
    <nav
      class="sc-kEjbQP kzyaRm"
    >
      <h1>
        <a
          href="/"
        >
          Seed
        </a>
      </h1>
      <nav>
        <ul>
          <li>
            <a
              class="link"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              class="link"
              href="/upload"
            >
              Upload
            </a>
          </li>
          <li>
            <a
              class="link"
              href="/mypage"
            >
              My list
            </a>
          </li>
        </ul>
      </nav>
    </nav>
    <section
      class="sc-iqHYmW iNMKRi"
    >
      <header
        class="sc-fubCzh heZVao"
      >
        <div
          class="sc-jrAFXE flZrjl"
        >
          <form
            class="sc-gsTEea dHwVSw"
          >
            <input
              autocomplete="off"
              class="sc-bdfBQB sc-dlfnuX gpSnii kJTEFr"
              name="search"
              placeholder="Search..."
              type="text"
              value=""
            />
            <button
              aria-label="search"
              class="MuiButtonBase-root MuiIconButton-root search-icon"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </form>
          
        </div>
        <div
          class="sc-pGacB blvLIr"
        >
          <a
            class="link"
            href="/login"
          >
            Login
          </a>
          <a
            class="link"
            href="/signup"
          >
            Signup
          </a>
        </div>
      </header>
      <div />
    </section>
  </div>
</div>
`;
